AI/Nullkiller/AIUtility.cpp

```diff
@@ -421,9 +421,16 @@ bool shouldVisit(const Nullkiller * ai, const CGHeroInstance * h, const CGObject
 	case Obj::BOAT:
 	case Obj::SIGN:
 		return false;
+	case Obj::SHRINE_OF_MAGIC_GESTURE:
+	case Obj::SHRINE_OF_MAGIC_INCANTATION:
+		return h->hasSpellbook();
+	case Obj::SHRINE_OF_MAGIC_THOUGHT:
+		return h->hasSpellbook() && h->getSecSkillLevel(SecondarySkill::WISDOM) >= MasteryLevel::BASIC;
+	case Obj::PYRAMID:
+		return h->hasSpellbook() && h->getSecSkillLevel(SecondarySkill::WISDOM) == MasteryLevel::EXPERT;
```

Comment:
Kind of wish to have this somewhere inside rewardable object...

Reply:
I think calling `getAvailableRewards(h, Rewardable::EEventType::EVENT_FIRST_VISIT);` should tell whether it makes sense for AI to visit this object, but needs tests

Reply:
Ok will take a look onto your suggestion

Reply:
getAvailableRewards is protected, but exactly what I did in other places except EEventType. Have not tested it, just hero limiter. So getAvailableRewards can be used if we make it public

Reply:
So can I make it public?

Reply:
Yeah, sure. 
I usually mark stuff as private / protected to indicate that it is not used outside of the class, so generally safe to make public.